File: cipher-codes/Base64Cipher.py
import base64

def encode_base64(message):
    message_bytes = message.encode('utf-8')
    base64_bytes = base64.b64encode(message_bytes)
    base64_message = base64_bytes.decode('utf-8')
    return base64_message

def decode_base64(base64_message):
    base64_bytes = base64_message.encode('utf-8')
    message_bytes = base64.b64decode(base64_bytes)
    message = message_bytes.decode('utf-8')
    return message

def main():
    mode = input("Do you want to encode or decode? (e/d): ").strip().lower()

    if mode == 'e':
        message = input("Enter the message to encode: ")
        encoded_message = encode_base64(message)
        print("Encoded message:", encoded_message)
    elif mode == 'd':
        base64_message = input("Enter the Base64 encoded message to decode: ")
        decoded_message = decode_base64(base64_message)
        print("Decoded message:", decoded_message)
    else:
        print("Invalid option. Please choose 'e' to encode or 'd' to decode.")

if __name__ == "__main__":
    main()

File: cipher-codes/Base64Cipher.java
import java.util.Base64;
import java.util.Scanner;

public class Base64Cipher {

    public static String encodeBase64(String message) {
        byte[] messageBytes = message.getBytes();
        byte[] base64Bytes = Base64.getEncoder().encode(messageBytes);
        return new String(base64Bytes);
    }

    public static String decodeBase64(String base64Message) {
        byte[] base64Bytes = base64Message.getBytes();
        byte[] messageBytes = Base64.getDecoder().decode(base64Bytes);
        return new String(messageBytes);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Do you want to encode or decode? (e/d): ");
        String mode = scanner.nextLine().trim().toLowerCase();

        if (mode.equals("e")) {
            System.out.print("Enter the message to encode: ");
            String message = scanner.nextLine();
            String encodedMessage = encodeBase64(message);
            System.out.println("Encoded message: " + encodedMessage);
        } else if (mode.equals("d")) {
            System.out.print("Enter the Base64 encoded message to decode: ");
            String base64Message = scanner.nextLine();
            String decodedMessage = decodeBase64(base64Message);
            System.out.println("Decoded message: " + decodedMessage);
        } else {
            System.out.println("Invalid option. Please choose 'e' to encode or 'd' to decode.");
        }

        scanner.close();
    }
}
